integration/map_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/merkle"
+	_ "github.com/google/trillian/merkle/coniks" // Register
+	"github.com/google/trillian/merkle/hashers"
+	_ "github.com/google/trillian/merkle/maphasher" // Register
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
+)
+
+// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
+// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
+func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
+	if len(iv)%2 != 0 {
+		panic(fmt.Sprintf("integration: createMapLeaves got odd number of iv pairs: %v", len(iv)))
+	}
+	r := []*trillian.MapLeaf{}
+	var index []byte
+	for i, b := range iv {
```

Comment:
Suggestion: I think writing a for with 2-step loops would make this clearer:

```go
	for i := 0; i < len(iv); i += 2 {
		r = append(r, &trillian.MapLeaf{
			Index:     iv[i],
			LeafValue: iv[i+1],
		})
	}
```

It took me a few moments to realize what you have is equivalent to the above.